occurrences/tests/test_api.py

```diff
@@ -2343,8 +2345,36 @@ def test_occurrences_filter_by_time(
     snapshot.assert_match(executed)
 
 
-@pytest.mark.parametrize("enrolment_end_days,count", [(None, 3), (0, 3), (1, 2)])
 def test_occurrences_filter_by_upcoming(
+    snapshot, api_client, mock_get_event_data, mock_update_event_data,
+):
+    p_event_1 = PalvelutarjotinEventFactory(
+        enrolment_start=datetime(2020, 1, 5, 0, 0, 0, tzinfo=timezone.now().tzinfo),
+    )
+
+    OccurrenceFactory(
+        start_time=datetime(2020, 1, 7, 0, 0, 0, tzinfo=timezone.now().tzinfo),
+        p_event=p_event_1,
+    )
+
+    OccurrenceFactory(
+        start_time=datetime(2020, 1, 6, 0, 0, 0, tzinfo=timezone.now().tzinfo),
+        p_event=p_event_1,
+    )
+
+    # Past occurrences
+    OccurrenceFactory(
+        start_time=datetime(2020, 1, 5, 0, 0, 0, tzinfo=timezone.now().tzinfo),
+        p_event=p_event_1,
+    )
+
+    executed = api_client.execute(OCCURRENCES_QUERY, variables={"upcoming": True})
+    snapshot.assert_match(executed)
+    assert len(executed["data"]["occurrences"]["edges"]) == 3
```

Comment:
Personally in tests like this, I like to check that the returned instances are the ones we are expecting to make sure the tests are not passing falsely.

Also, isn't this returning all the occurrences (3), but one is in the past and should not be returned I assume?

Reply:
Yea, sorry that "past occurrence" comment was left there from a copy-paste. 👎 3 is the right result. There is a snapshot assertion that checks the right objects. Because this test does not have a database connection, I cannot test occurrence ids. I cannot compare them directly either, because the other one is an graphql document and another one is an ORM instance. The count test is there to make the test clearer and easier to understand, but it didn't work this time I guess :D 

If really wanted, I can remove the snapshot test or the count tes and compare dates or something, but I don't see that it would be good to test the both, the snapshot and the dates.

Reply:
Blah, I managed to miss that there is also a snapshot in this test, my bad.